### PR TITLE
Remove `BASE_DIR` parsing for settings

### PR DESCRIPTION
**Pull request by b-rowan**
`settings.yaml` being parsed from a base directory has issues when being installed from pip.  The user must now pass a `settings_file` path to goosebit when running.

With these changes, gooseBit should be run with `python -m goosebit [settings_file]`.  There is also now help information with `python -m goosebit --help`.

Fixes: #103

### CODE REVIEW

**Comment by b-rowan** [comment]
Also note, `settings_file` has to be optional, because tests will break otherwise.  All settings should have a valid default.

File: goosebit/settings/schema.py [hunk 1]
```diff
@@ -61,9 +63,17 @@ def settings_customise_sources(
...
     ) -> tuple[PydanticBaseSettingsSource, ...]:
+        # skip arg parsing if running pytest
+        if "pytest" not in sys.modules:
+            cli_args = parser.parse_args()
+            return (
+                init_settings,
+                YamlConfigSettingsSource(settings_cls, cli_args.settings_file),
+                env_settings,
+                file_secret_settings,
+            )
```
Comment (b-rowan) [inline goosebit/settings/schema.py:74]: Not a fan of this personally, but I cant think of a better option with the current project structure...

File: goosebit/settings/args.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import argparse
```
Comment (easybe) [inline goosebit/settings/args.py:1]: This is not where I would expect argument parsing. IMHO this should be in `__main__.py`.

**Review by easybe — COMMENTED** [review]
I don't think this is going into the right direction. What if you want to run the service with e.g. `gunicorn` together with a settings file?

I would rather see the app search for a system-wide file (e.g. in `/etc`) and maybe the current directory. This can be made portable with e.g. the `platformdirs` package.

File: goosebit/settings/schema.py (hunk 1 above)
Comment (easybe) [reply goosebit/settings/schema.py:74]: This looks awful.

**Comment by b-rowan** [comment]
> I don't think this is going into the right direction. What if you want to run the service with e.g. `gunicorn` together with a settings file?
> 
> I would rather see the app search for a system-wide file (e.g. in `/etc`) and maybe the current directory. This can be made portable with e.g. the `platformdirs` package.

Fair enough.  Closing because I dont think this makes much sense then

**Pull request closed without merging**